writer.go

```diff
@@ -727,53 +724,77 @@ func (w *writer) dial() (conn *Conn, err error) {
 	return
 }
 
-func (w *writer) write(conn *Conn, batch []Message, resch [](chan<- error)) (ret *Conn, err error) {
-	if conn == nil {
-		if conn, err = w.dial(); err != nil {
-			w.stats.errors.observe(1)
-			w.withErrorLogger(func(logger *log.Logger) {
-				logger.Printf("error dialing kafka brokers for topic %s (partition %d): %s", w.topic, w.partition, err)
-			})
-			for i, res := range resch {
-				res <- &writerError{msg: batch[i], err: err}
-			}
-			return
-		}
+func shouldRetry(err error, retries, attempts int) bool {
+	if retries == attempts {
```

Comment:
`if retries >= attempts` is a little safer in the event that `retries` was accidentally incremented twice

Reply:
yep, changed to that and updated some names to make it more clear